### PR TITLE
Further changes to the new `shell` package

### PR DESCRIPTION
**Pull request by aalexandrov**
- Use entry_points-based name -> type resolution of the forecaster type in the CLI.
- Mark the module as `.typesafe`.

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-173/1 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-173/1/index.html

**Comment by szha** [comment]
Job PR-173/3 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-173/3/index.html

File: src/gluonts/shell/sagemaker/__init__.py [hunk 1]
```diff
@@ -36,18 +37,16 @@ def __init__(self, path="/opt/ml"):
         self.datasets = self._get_datasets()
         self._check_sf2()
 
-    def _get_datasets(self):
-        freq = self.hyperparameters["time_freq"]
+    def _get_datasets(self) -> Dict[str, FileDataset]:
+        freq = self.hyperparameters["freq"]
```
Comment (lostella) [inline src/gluonts/shell/sagemaker/__init__.py]: So this won't break anything?

**Comment by szha** [comment]
Job PR-173/4 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-173/4/index.html

**Comment by szha** [comment]
Job PR-173/5 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-173/5/index.html

File: src/gluonts/shell/sagemaker/__init__.py (hunk 1 above)
Comment (aalexandrov) [reply src/gluonts/shell/sagemaker/__init__.py]: Good catch. I cleaned up all SageMaker-related configuration and environment bootstrapping logic (now all in `sagemaker/__init__.py`) and updated the PR.

The following method now ensures backwards-compatibility w.r.t. the `hyperparameters.json` file. Backwards-compatibility in the `metadata.json` file was already added in the current version.

```python
def _load_hyperparameters(path: MLPath, channels) -> dict:
    with path.hyperparameters.open() as json_file:
        hyperparameters = parse_sagemaker_parameters(json.load(json_file))

        for old_freq_name in ['time_freq', 'time_granularity']:
            if old_freq_name in hyperparameters:
                hyperparameters['freq'] = hyperparameters[old_freq_name]

        if "metadata" in channels:
            with (channels["metadata"] / "metadata.json").open() as file:
                metadata = MetaData(**json.load(file))
                hyperparameters.update(freq=metadata.freq)

        return hyperparameters
```

**Comment by szha** [comment]
Job PR-173/6 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-173/6/index.html

**Comment by lostella** [comment]
It would be nice to add tests for the `gluonts.shell` package, but this can be done in a separate PR: I'm opening an issue to track this

**Pull request merged**